queries.sql

```diff
@@ -1,3 +1,44 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-SELECT * from animals WHERE name = 'Luna';
\ No newline at end of file
+SELECT * from animals WHERE name = 'Luna';
+SELECT * FROM animals WHERE name LIKE '%mon';
+SELECT name FROM animals WHERE date_of_birth >= '2016-01-01' AND date_of_birth < '2019-01-01';
+SELECT name FROM animals WHERE neutered = 't' AND escape_attempts < 3;
+SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR 'Pikachu';
+SELECT  name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+SELECT * from animals WHERE neutered = 'true';
+SELECT * FROM animals WHERE name != 'Gabumon';
+SELECT * FROM animals WHERE weight_kg > 10.4 AND weight_kg =< 17.3;
+
+/* task-2 */
+BEGIN;
+   UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+   UPDATE animals SET species = 'pokemon' WHERE species = null;
+COMMIT;
+  
+BEGIN;
+   DELETE FROM animals;
+   ROLLBACK;
+COMMIT;
+
+BEGIN;
+   DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+   SAVEPOINT sp1;
+   UPDATE animals SET weight_kg = weight_kg * -1 ;
+   ROLLBACK TO sp1;
+   UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT SUM(escape_attempts), neutered FROM animals GROUP BY neutered;
+/* What is the minimum and maximum weight of each type of animal? */
+SELECT MAX(weight_kg),MIN(weight_kg), neutered FROM animals GROUP BY neutered;
+/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
+SELECT AVG(escape_attempts), neutered FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01' GROUP BY neutered ;
```

Comment:
* I think you can use the species column added to the animal table to write our queries 
* For example for this question ( What is the minimum and maximum weight of each type of animal? ) answer (
` SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;`  here we use species for differentiating animal types 
``` )

Reply:
:heavy_check_mark: Done!

Reply:
* This hasn't been fixed you are still selecting animals grouped by neutered while you should use species as i explained above this should be your answer for this question ( What is the minimum and maximum weight of each type of animal?)  `SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;`
* This is because the species column is used to specify the animal type 
* For this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?) you should add `SELECT species and then group by species` to your answer this way your query is targeting a specific animal type whose birthdate falls between `1990 to 2000`

Reply:
Seems like I didn't push the updated branch. I edited it to display : 
```cmd
/* What is the minimum and maximum weight of each type of animal? */
SELECT MAX(weight_kg),MIN(weight_kg), species_id FROM animals GROUP BY species_id;

/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
SELECT AVG(escape_attempts), species_id  FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01' GROUP BY species_id ;
```

Hope it is ok.